### ISSUE TITLE
Support for other internal variables than ${buildType} in cmake.buildDirectory

### ISSUE THREAD

**Issue by vogel76** [vector-of-bool/vscode-cmake-tools#272]
CmakeTools has great feature which allows to separately define settings for buildType (debug, release etc) and also for project variants by `cmake.defaultVariants` definition. Then it is possible to choose all their combinations to build.
For complex projects where such feature is useful, it could be convenient to make `cmake.buildDirectory` value also dependent on chosen `buildType` and projectVariant.
Let's allow to reference all variants defined in cmake.defaultVariants object as variables, similary to buildType.
Example:
```
"cmake.defaultVariants": {
   "buildType": {
      "default$": "debug",
      "description$": "the build type to use",
      "debug": {
         "oneWordSummary$": "DebugClang",
         "description$": "Emit debug information without performing optimizations",
         "buildType": "Debug",
         "settings": {
         "CMAKE_CXX_COMPILER":"clang++",
         "CMAKE_C_COMPILER": "clang"
         }
      },
      "release": {
         "oneWordSummary$": "Release",
         "description$": "Enable optimizations, omit debug info",
         "buildType": "Release"
      },
      "reldeb": {
         "oneWordSummary$": "RelWithDebInfo",
         "description$": "Perform optimizations AND include debugging information",
         "buildType": "RelWithDebInfo"
      }
   },
   "projectVariant": {
      "default$": "FreeVersion",
      "description$": "Free version of application",
      "FreeVersion": {
         "oneWordSummary$": "Free",
         "description$": "Settings specific to free version of app",
         "settings": {
            "ENABLE_PAID_FEATURES": "OFF"
          }
      },
      "PaidVersion": {
         "oneWordSummary$": "Paid",
         "description$": "Settings specific to paid version of app",
         "settings": {
            "ENABLE_PAID_FEATURES": "ON"
          }
      }
   }
   },
   "cmake.buildDirectory": "${workspaceRoot}/build/${buildType}/${projectVariant}"
```
For example, after project configuration buildDirectory will point to <br/> `<workspace-root>/build/Debug/PaidVersion`

**Comment by vector-of-bool** [comment]
Thanks for the issue and PR. I'm hoping to finish a big refactor by the end of the year, and then I can jump on things like this. Sorry for the delay.

**Comment by vogel76** [comment]
Ok, thx for the info. May I know which changes do you plan to do (ie new functionality) ?

**Comment by stayprivates** [comment]
Hi,

Any progress on this issue?  Does the .0.11 beta4 has that feature.  I could not find anything in the documentation about it.

**Comment by vogel76** [comment]
Hi,

                Just resolved conflicts and branch and PR are ready for your review and merge.

 

Best regards,

                Vogel

**Comment by Randshot** [comment]
Fixed in #317.
**Closed**.
If this issues still persists in the current **_0.11.0-beta4_**, feel free to reopen this one or create a new one.

**Comment by Randshot** [comment]
PR #317 merged.
